src/data-transport.ts

```diff
@@ -58,18 +59,17 @@ export default class DataTransport {
       headers
     }
 
-    this._sendData(options, metrics)
+    this._logger.trace('Sending metrics:', options, headers, metrics)
+    this._sendData(options, metrics, cb)
   }
 
-  private _sendData(sendOptions: SendDataOptions, payload: string): void {
-    // TODO: control flow for shutdown case & errors
+  private _sendData(sendOptions: SendDataOptions, payload: string, callback: any): void {
```

Comment:
Instead of `any` callback type, let's get specific about the contract.

```suggestion
  private _sendData(sendOptions: SendDataOptions, payload: string, callback: (error: Error) => void): void {
```